cli/azd/cmd/middleware/telemetry.go

```diff
@@ -65,6 +69,11 @@ func (m *TelemetryMiddleware) Run(ctx context.Context, next NextFn) (*actions.Ac
 
 	span.SetAttributes(fields.CmdArgsCount.Int(len(m.options.Args)))
 
+	// Set the platform type when available
+	if platformConfig, err := m.lazyPlatformConfig.GetValue(); err == nil && platformConfig != nil {
+		span.SetAttributes(fields.PlatformTypeKey.String(string(platformConfig.Type)))
```

Comment:
For any logged telemetry, we will need to ensure that no user data is logged. The data logged must:

1. Be guaranteed to be a system-value, and not an arbitrary user value.
2. Alternatively, we always hash the user-provided value

Is it possible that `platformConfig.Type` is set to `some-random-value` here?

Reply:
This value is set from configuration which is user supplied.  Yes, it could be possible that a user could enter a random value using `azd config set platform.type <random>` but that is never valid or expected.

Reply:
We do want to safeguard against that, or to always hash the value to be consistent with our current data guidelines.

Reply:
Will wait for this to be addressed before final approval.

Reply:
I'm thinking we compare this value against an allowed values and only hash the value if is not expected.  That should give us good query support and also protect against any arbitrary values.

Reply:
Updated to include validation of platform type to only allow allowed values.  The telemetry will now never send a non-supported value.